modyn/config/schema/pipeline/trigger/drift/detection_window/amount.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from functools import cached_property
+from typing import Annotated, Literal, Union
+
+from modyn.config.schema.base_model import ModynBaseModel
+from modyn.const.regex import REGEX_TIME_UNIT
+from modyn.utils.utils import SECONDS_PER_UNIT
+from pydantic import Field
+
+
+class _BaseWindowingStrategy(ModynBaseModel):
+    allow_overlap: bool = Field(
+        False,
+        description=(
+            "Whether the windows are allowed to overlap. This is useful for time-based windows."
+            "If set to False, the current window will be reset after each trigger."
+        ),
+    )
+
+
+class AmountWindowingStrategy(_BaseWindowingStrategy):
+    id: Literal["AmountWindowingStrategy"] = Field("AmountWindowingStrategy")
+    amount_ref: int = Field(1000, description="How many data points should fit in the reference window", ge=1)
+    amount_cur: int = Field(1000, description="How many data points should fit in the current window", ge=1)
+
+    @property
+    def current_buffer_size(self) -> int | None:
+        return self.amount_cur
```

Comment:
The model is used as a Union of several `WindowingStrategies`. Some don't let the user choose this field and, therefore, return None. The None here is for type consistency. Arguably it might also work to return the narrowest type and hope that the Union type resolves the property to a Union return value.